components/button/button-toggle.js

```diff
@@ -0,0 +1,83 @@
+import { css, html, LitElement } from 'lit';
+
+/**
+ * A button container component for button toggles.
+ */
+class ButtonToggle extends LitElement {
+
+	static get properties() {
+		return {
+			/**
+			 * Pressed state
+			 * @type {boolean}
+			 */
+			pressed: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			::slotted(:not(d2l-button-icon, d2l-button-subtle)),
+			:host slot[name="pressed"],
+			:host([pressed]) slot[name="not-pressed"] {
+				display: none;
+			}
+			:host slot[name="not-pressed"],
+			:host([pressed]) slot[name="pressed"] {
+				display: contents;
+			}
+		`;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (this._focusOnFirstRender) {
+			this._focusOnFirstRender = false;
+			this.focus();
+		}
+	}
+
+	render() {
+		return html`
+			<slot @click="${this._handleNotPressedClick}" name="not-pressed"></slot>
+			<slot @click="${this._handlePressedClick}" name="pressed"></slot>
+		`;
+	}
+
+	focus() {
+		if (!this.hasUpdated) {
+			this._focusOnFirstRender = true;
+			return;
+		}
+
+		const elem = this.shadowRoot.querySelector(this.pressed ? 'slot[name="pressed"]' : 'slot[name="not-pressed"]').assignedNodes()[0];
+		if (!elem) {
+			throw new Error('d2l-button-toggle: no button to focus');
+		}
+
+		elem.focus();
+	}
+
+	async _handleClick(pressed) {
+		this.pressed = pressed;
```

Comment:
Shouldn't the underlying `<button>`s get `aria-pressed="true"`

Reply:
Not necessarily. Jeff and I did a bunch of investigation into that. Using `aria-pressed` is one approach but the guidelines suggest that it's not required if the state is communicated via the label. Jeff preferred not using it due the confusion around the label and the state.

Reply:
Yeah agreed. I have a side conv with Jeff to craft something for this, since we'll want to provide some guidance around how they should set their labels.